PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/handler/IntelliJAzureExceptionHandler.java

```diff
@@ -211,6 +213,18 @@ private String getActionText(final List<Throwable> throwableList) {
         return registerAction == null ? actions : ArrayUtils.addAll(actions, registerAction);
     }
 
+    public static List<AzureOperationRef> revise(Deque<? extends AzureOperationRef> operations) {
+        final LinkedList<AzureOperationRef> result = new LinkedList<>();
+        for (final AzureOperationRef op : operations) {
```

Comment:
So we only track operation stack to the latest action? What if user action call another action, like create web app will call refresh explorer, which may hide the origin user action.

Reply:
This is a compromise solution to our current implementation, and I am actually considering if we should add another type of operation.

Think about the case: refreshing explorer fails after successfully create a web app. 
what should the message looks like? 
`failed to create web app, as refreshing explorer failed ?`
No, it's not the fact since webapp is actually successfully created.

the problem is that. some operations are required part of the parent operation, and the others are not(optional or just side effects) e.g. 
in our case/implementation of web app creation. 
```
USER ACTION(create web app) {
    1. SERVICE(check login status/login)
    2. "SERVICE(open dialog)"
      3. SERVICE(create web app)
        4. ACTION(deploy artifact)
        5. ACTION(refresh explorer)
}
```
1 & 2 & 3 are required parts of `create web app`, the failure of these operations will cause failure of `create webapp`.
but 4 & 5 are not the require parts, the failure of these operations will **NOT** cause failure of `create webapp`.

as a compromise to our current design/implementation, I make these optional/side effect operations `ACTION`.
I have come up with a very raw idea to avoid this kind of compromise. let's discuss it when you come back.